lms/services/grouping.py

```diff
@@ -52,30 +53,20 @@ def upsert_with_parent(  # pylint: disable=too-many-arguments
             tool_consumer_instance_guid, lms_id, parent, type_
         )
 
-        grouping = (
-            self._db.query(Grouping)
-            .filter_by(
-                application_instance=self._application_instance,
-                authority_provided_id=authority_provided_id,
-            )
-            .one_or_none()
+        return upsert(
+            self._db,
+            Grouping,
+            query_kwargs={
```

Comment:
Not sure about the API, the same param controls both the query and the creation of a new object. Sometimes the query would be equivalent but sometimes that many params might confuse the query planner.

Spiting it into three arguments (query, create_args, update_args?) could be nicer but not sure if worth the effort over just doing it in SQLA directly in the service.